File: php/classes/constructors_and_destructors/constructor_with_arguments.php
#!/usr/bin/php8.1
<?php

class Basket {
  public function __construct(string $label, int $capacity) {
    $this->label = $label;
    $this->capacity = $capacity;
    $this->load_counter = 0;
  }

  public function put(string $item): void {
    if ($this->load_counter < $this->capacity) {
      $this->items[$this->load_counter] = $item;
    }
    $this->load_counter++;
  }

  public function pull(): string {
    $this->item = $this->items[$this->load_counter];
    $this->items[$this->load_counter] = "";
    $this->load_counter--;

    return $this->item;
  }

  public function show(): void {
    print($this->label . ": ");
    foreach ($this->items as $item) {
      print($item . " ");
    }
    print("\n");
  }

  private string $label;
  private int $capacity;
  private array $items = [];
  private int $load_counter;
}

$handbasket = new Basket("fruits", 5);

$handbasket->put("apple");
$handbasket->put("pear");
$handbasket->put("plum");
$handbasket->put("peach");
$handbasket->put("grape");

$handbasket->show();

File: python/classes/constructors_and_destructors/constructor_with_arguments.py
#!/usr/bin/python3

class Basket:
    def __init__(self, label, capacity):
        self.label = label
        self.capacity = capacity
        self.items = []
        self.load_counter = 0

    def put(self, item):
        if (self.load_counter < self.capacity):
            self.items.append(item)
            self.load_counter += 1

    def pull(self):
        if (self.load_counter > 0):
            item = self.items[self.load_counter - 1]
            self.items[self.load_counter - 1] = ""
            self.load_counter -= 1
        else:
            item = ""

        return item

    def show(self):
        print(self.label + ": ", end = "")
        for item in self.items:
            print(item + " ", end = "")

        print()

handbasket = Basket("fruits", 5)

handbasket.put("apple")
handbasket.put("pear")
handbasket.put("plum")
handbasket.put("peach")
handbasket.put("grape")

print("BASKET:")

handbasket.show()

print("REMOVED ITEMS:")

print(handbasket.pull())
print(handbasket.pull())
print(handbasket.pull())

print("BASKET:")

handbasket.show()

print("REMOVED ITEMS:")

print(handbasket.pull())
print(handbasket.pull())
print(handbasket.pull())

print("BASKET:")

handbasket.show()
